rust/src/lib/route.rs

```diff
@@ -77,6 +79,22 @@ impl Routes {
                     error!("{}", e);
                     return Err(e);
                 }
+                if let Some(dst) = route.destination.as_ref() {
+                    let dst_str = dst.as_str();
+                    if !is_ipv6_addr(dst_str)
+                        && !is_route_dst_valid(dst_str).unwrap_or_default()
```

Comment:
You have two layer of raising errors. And this `!foo().unwrap_or_default()` is hurting human eyes.

Try change to something like:


```rust
if let Some(dst) = route.destination.as_deref() {
    validate_route_dst(dst)?;
}


fn validate_route_dst(dst: &str) -> Result<(), NmstateError> {
    todo!()
}
```